nexus/src/app/background/tasks/instance_watcher.rs

```diff
@@ -377,15 +377,23 @@ impl BackgroundTask for InstanceWatcher {
     ) -> BoxFuture<'a, serde_json::Value> {
         async {
             let mut tasks = tokio::task::JoinSet::new();
-            let mut paginator = Paginator::new(MAX_SLED_AGENTS);
-            let mk_client = |sled: &Sled| {
-                nexus_networking::sled_client_from_address(
-                    sled.id(),
-                    sled.address(),
-                    &opctx.log,
-                )
-            };
+            let mut paginator = Paginator::new(MAX_CONCURRENT_CHECKS);
+
+            let mut total: usize = 0;
+            let mut update_sagas_queued: usize = 0;
+            let mut instance_states: BTreeMap<String, usize> =
+                BTreeMap::new();
+            let mut check_failures: BTreeMap<String, usize> =
+                BTreeMap::new();
+            let mut check_errors: BTreeMap<String, usize> = BTreeMap::new();
 
+            // A `reqwest` client is a reference-counted handle to a connection
+            // poll that can be reused by multiple requests. Making a new client
+            // is fairly expensive, but cloning one is cheap, and cloning it
+            // allows reusing pooled TCP connections. Therefore, we will order
+            // the database query by sled ID, and reuse the same sled-agent
+            // client as long as we are talking to the same sled.
```

Comment:
This is fine to do here for now I think we really want to be doing this using q-orb at the `Resolver` level or equivalent and I think that's at least nominally the plan.

Reply:
Totally, and when we're using Qorb for sled-agent clients, we can probably simplify this code a bit!